sdk/tools/azure-sdk-archetype/README.md

```diff
@@ -0,0 +1,31 @@
+# Azure SDK Maven Archetype
+
+The Azure SDK archetype can accelerate the bootstrapping of a new project that uses Azure SDK by 
+creating an application structure that follows many of the recommended best practices for using the Azure SDK for Java. In particular, the Azure SDK for Java Maven archetype creates a new Maven project with the following features:
+
+* A dependency on the latest `azure-sdk-bom` BOM release, to ensure that all Azure SDK for Java dependencies are aligned and give you the best developer experience possible.
+* Built-in support for GraalVM native image compilation.
+* Support for generating a new project with a specified set of Azure SDK for Java client libraries.
+* Integration with the Azure SDK for Java build tooling that will give build-time analysis of your project to ensure as many best practices are followed.
+
+As the Azure SDK for Java Maven archetype is published to Maven Central, to use it to create a new project you should enter the following into your terminal:
+
+```sh
+ mvn archetype:generate                        \
+   -DarchetypeGroupId=com.azure.tools          \
+   -DarchetypeArtifactId=azure-sdk-archetype \
```

Comment:
I believe it is required that we have the version specified here. Can you please double check (I've updated the draft text for docs.microsoft.com in my PR, but I'd appreciate a double-check on this, because it would be nice to not have to version in the readme and docs)

Reply:
It worked for me without specifying the version:

```
$ mvn archetype:generate -DarchetypeGroupId=com.azure.tools -DarchetypeArtifactId=azure-maven-archetype
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------< org.apache.maven:standalone-pom >-------------------
[INFO] Building Maven Stub Project (No POM) 1
[INFO] --------------------------------[ pom ]---------------------------------
[INFO]
[INFO] >>> maven-archetype-plugin:3.2.0:generate (default-cli) > generate-sources @ standalone-pom >>>
[INFO]
[INFO] <<< maven-archetype-plugin:3.2.0:generate (default-cli) < generate-sources @ standalone-pom <<<
[INFO]
[INFO]
[INFO] --- maven-archetype-plugin:3.2.0:generate (default-cli) @ standalone-pom ---
[INFO] Generating project in Interactive mode
[INFO] Archetype [com.azure.tools:azure-maven-archetype:1.0.0-SNAPSHOT] found in catalog local
Define value for property 'groupId': com.test.azure
Define value for property 'artifactId': azure-test-app
Define value for property 'version' 1.0-SNAPSHOT: :
Define value for property 'package' com.test.azure: :
Define value for property 'azureLibraries': azure-security-keyvault-keys,azure-identity
[INFO] Using property: enableGraalVM = true
[INFO] Using property: javaVersion = 17
[INFO] Using property: junitVersion = 5
Confirm properties configuration:
groupId: com.test.azure
artifactId: azure-test-app
version: 1.0-SNAPSHOT
package: com.test.azure
azureLibraries: azure-security-keyvault-keys,azure-identity
enableGraalVM: true
javaVersion: 17
junitVersion: 5
 Y: :
[INFO] ----------------------------------------------------------------------------
[INFO] Using following parameters for creating project from Archetype: azure-maven-archetype:1.0.0-SNAPSHOT
[INFO] ----------------------------------------------------------------------------
[INFO] Parameter: groupId, Value: com.test.azure
[INFO] Parameter: artifactId, Value: azure-test-app
[INFO] Parameter: version, Value: 1.0-SNAPSHOT
[INFO] Parameter: package, Value: com.test.azure
[INFO] Parameter: packageInPathFormat, Value: com/test/azure
[INFO] Parameter: junitVersion, Value: 5
[INFO] Parameter: package, Value: com.test.azure
[INFO] Parameter: javaVersion, Value: 17
[INFO] Parameter: groupId, Value: com.test.azure
[INFO] Parameter: artifactId, Value: azure-test-app
[INFO] Parameter: azureLibraries, Value: azure-security-keyvault-keys,azure-identity
[INFO] Parameter: enableGraalVM, Value: true
[INFO] Parameter: version, Value: 1.0-SNAPSHOT
[WARNING] Don't override file C:\Users\srnagar\home\azure\code\archetype-test\azure-test-app\src\main\java\com\test\azure
[WARNING] Don't override file C:\Users\srnagar\home\azure\code\archetype-test\azure-test-app\src\test\java\com\test\azure
[INFO] Executing META-INF/archetype-post-generate.groovy post-generation script
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/C:/Users/srnagar/.m2/repository/org/codehaus/groovy/groovy/2.4.16/groovy-2.4.16.jar) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Updating to latest versions...
[INFO] Scanning for projects...
[INFO]
[INFO] -------------------< com.test.azure:azure-test-app >--------------------
[INFO] Building azure-test-app 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- versions-maven-plugin:2.8.1:update-properties (default-cli) @ azure-test-app ---
[INFO] artifact com.azure:azure-sdk-bom: checking for updates from central
[INFO] Major version changes allowed
[INFO] Updated ${bom.version} from 1.0.2 to 1.0.5
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.982 s
[INFO] Finished at: 2021-11-17T13:35:37-08:00
[INFO] ------------------------------------------------------------------------
[INFO] Project created from Archetype in dir: C:\Users\srnagar\home\azure\code\archetype-test\azure-test-app
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:01 min
[INFO] Finished at: 2021-11-17T13:35:37-08:00
[INFO] ------------------------------------------------------------------------

```